packages/headless/src/controllers/insight/triggers/headless-insight-notify-trigger.ts

```diff
@@ -0,0 +1,35 @@
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {logNotifyTrigger} from '../../../features/triggers/insight-trigger-analytics-actions';
+import {
+  buildCoreNotifyTrigger,
+  NotifyTrigger,
+  NotifyTriggerState,
+} from '../../core/triggers/headless-core-notify-trigger';
+
+export type {NotifyTrigger, NotifyTriggerState};
+
+/**
+ * Creates an insight `NotifyTrigger` controller instance.
+ *
+ * @param engine - The insight engine.
+ * @returns A `NotifyTrigger` controller instance.
+ * */
+export function buildNotifyTrigger(engine: InsightEngine): NotifyTrigger {
```

Comment:
From what I can tell, the only thing that differs between the insight notify trigger controller and the search notify trigger controller is that they dispatch different analytics actions.

Instead of overriding the `subscribe` method, I think you could pass the analytics action creator as a prop to the core controller builder, and then just return the controller, i.e.,

```typescript
// headless-core-notify-trigger.ts

// ...
export interface NotifyTriggerProps {
  options: NotifyTriggerOptions
}

export interface NotifyTriggerOptions {
  logNotifyTriggerActionCreator: () => Action;
}

export function buildCoreNotifyTrigger(engine: CoreEngine, props: NotifyTriggerProps): NotifyTrigger {
  const {logNotifyTrigger} = props.options.logNotifyTriggerActionCreator;
  // ...
}
```

```typescript
// headless-insight-notify-trigger.ts

//...
import {logNotifyTrigger} from '../../../features/triggers/insight-trigger-analytics-actions';
// ...

export function buildNotifyTrigger(engine: InsightEngine): NotifyTrigger {
  return buildCoreNotifyTrigger(engine, {
    options: {
      logNotifyTriggerActionCreator: logNotifyTrigger,
    },
  });
}
```

```typescript
// headless-notify-trigger.ts

//...
import {logNotifyTrigger} from '../../features/triggers/trigger-analytics-actions';
// ...

export function buildNotifyTrigger(engine: SearchEngine): NotifyTrigger {
  return buildCoreNotifyTrigger(engine, {
    options: {
      logNotifyTriggerActionCreator: logNotifyTrigger,
    },
  });
}
```

Reply:
yeah good idea!